skorch/scoring.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+from skorch.net import NeuralNet
+from skorch.dataset import unpack_data
+
+
+def loss_scoring(net: NeuralNet, X, y=None):
+    """
+    loss_scoring(net, X, y=None)
+
+    Computes the loss of ``net`` on data (``X``, ``y``).
+
+    This function can be used to implement the ``score`` method for a
+    :class:`.NeuralNet` through sub-classing. This is useful, for example, when
+    combining skorch models with sklearn objects that rely on the model's
+    ``score`` method. Below is an example using GridSearchCV.
+
+    >>> class ScoredNet(skorch.NeuralNetClassifier):
+    ...     def score(self, X, y=None, lower_is_better=False):
```

Comment:
```suggestion
    ...     def score(self, X, y=None):
```

I would remove this because it doesn't conform to the sklearn API. Also, if `lower_is_better = True`, which is always true for PyTorch criteria, we should return `-loss_value` and not vice versa? Otherwise, something like grid search would choose the model with the _highest_ loss.